internal/v2/fakes2av2/fakes2av2_test.go

```diff
@@ -0,0 +1,177 @@
+package fakes2av2
+
+import (
+	"net"
+	"log"
+	"time"
+	"sync"
+	"context"
+	"testing"
+	"google.golang.org/grpc"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	s2av2pb "github.com/google/s2a-go/internal/proto/v2/s2a_go_proto"
+	commonpb "github.com/google/s2a-go/internal/proto/v2/common_go_proto"
+)
+
+const (
+	defaultTimeout = 30.0 * time.Second
+)
+
+func startFakeS2Av2Server(wg *sync.WaitGroup) (address string, stop func(), err error) {
+	// Pick unused port.
+	listener, err := net.Listen("tcp", ":0")
+	address = listener.Addr().String()
+	if err != nil {
+		log.Fatalf("failed to listen on port %s: %v", listener.Addr().String(), err)
+	}
+	s := grpc.NewServer()
+	log.Printf("Server: started gRPC Fake S2Av2 Server at port: %s", listener.Addr().String())
```

Comment:
micro-nit: s/at port/on address

since listener.Addr().String() is the full address

Similarly below.